### PR TITLE
Remove ICU

### PR DESCRIPTION
**Pull request by mck1117**
Remove ICU completely, allowing only EXTI for trigger input (and COMP on Kinetis 😉), closing #639 and #630.

This change removes the ICU code, disables it in `mcuconf.h` and `halconf.h`, and fixes up usages of the ICU to use EXTI instead (frequency MAP and vehicle speed).  Trigger, VSS, and freq MAP have all been tested and work great.

Somehow this change also manages to save 17 kilobytes of flash space on a release build. Yep. 17k.

### CODE REVIEW

**Comment by rusefi** [comment]
did we run any performance comparison between icu and exti?
i have an interesting mental block about removing old code.

**Comment by rusefi** [comment]
did we run any performance comparison between icu and exti?
i have an interesting mental block about removing old code. 

@andreika-git what do you think?

File: firmware/development/wave_analyzer.cpp [hunk 1]
```diff
@@ -203,49 +178,45 @@ static float getSignalPeriodMs(int index) {
...
-	if (readers[index].hw->started) {
-//	int counter = getEventCounter(index);
-//	debugInt2(logging, "ev", index, counter);
 
-		float dwellMs = getSignalOnTime(index);
-		float periodMs = getSignalPeriodMs(index);
-
-		appendPrintf(logging, "duty%d%s", index, DELIMETER);
-		appendFloat(logging, 100.0f * dwellMs / periodMs, 2);
+	float dwellMs = getSignalOnTime(index);
```
Comment (mck1117) [inline firmware/development/wave_analyzer.cpp]: I'm not sure why it thought this was a diff, all I did was unwrap it from an `if (...)`

File: firmware/development/wave_analyzer.cpp [hunk 2]
```diff
@@ -266,12 +237,12 @@ void initWaveAnalyzer(Logging *sharedLogger) {
 #if EFI_WAVE_ANALYZER
 	initWave(PROTOCOL_WA_CHANNEL_1, 0);
 	initWave(PROTOCOL_WA_CHANNEL_2, 1);
+	initWave(PROTOCOL_WA_CHANNEL_3, 2);
+	initWave(PROTOCOL_WA_CHANNEL_4, 3);
```
Comment (mck1117) [inline firmware/development/wave_analyzer.cpp]: looks like these were just missing before?

**Comment by rusefi** [comment]
pins 0 thru 4 get their own interrupt

5-9 share one
10-15 share one

@mck1117 says
all the code has to do is check which mask bits are set in the IMR register (ie, bit 7 would be set if channel 7 caused the interrupt)
I looked at the disassembly and it's two cycles per channel
effectively
```
skip_next_if_bit_not_set(pr,5);
handle_exti_5();
skip_next_if_bit_not_set(pr,6);
handle_exti_6();
skip_next_if_bit_not_set(pr,7);
handle_exti_7();
```
...

**Comment by rusefi** [comment]
I would like to take a different approach on this one:

first, I would like to unify ICU and PAL implementations for those to provide same internal API, then I would like to keep ICU via compile-time defines. I know there might be different opinions on this but hopefully we can agree that this is not too important.

**Pull request closed without merging**